src/tree/App.js

```diff
@@ -12,42 +18,56 @@ class App extends EnhancedTreeItem {
 		this.public = isPublic
 		this.approved = isApproved
 		this.level = 0
-		this.contextValue = "app" + (this.approved ? "_approved" : this.public ? "_public" : "") + (changes !== undefined ? (changes.length !== 0 ? "_changed" : "") : "")
+		this.contextValue = "app"
+			+ (isApproved ? "_approved" : "")
+			+ (!isApproved && isPublic ? "_public" : "")
+			+ (changes?.length !== 0 ? "_changed" : "");
 		this.theme = theme
-		this.iconVersion = iconVersion
+		/** Defines the final png icon filename. */
+		this.iconVersion = iconVersion;
 		this.changes = changes
+		/** Is the app opensource? If app is opensource, then green square is not added to icon.  */
+		this.isOpensource = isOpensource;
 		this.tooltip = this.makeTooltip()
-		this.iconPath = this.makeIconPath(iconDir)
+		this.iconPath = this.makeIconPath();
 		this.rawIcon = {
-			dark: path.join(iconDir, `${this.name}.${this.iconVersion}.png`),
-			light: path.join(iconDir, `${this.name}.${this.iconVersion}.dark.png`)
+			dark: path.join(appsIconTempDir, `${this.name}.${this.iconVersion}.png`),
+			light: path.join(appsIconTempDir, `${this.name}.${this.iconVersion}.dark.png`)
 		}
 	}
-	makeIconPath(iconDir) {
-		if (!this.public) {
+
+	/**
+	 * Get path to icon file.
+	 * If appp is public, then used public icon (it means icon with small green square in right bottom corner).
+	 * @param {string} iconDir Basedir of icons.
+	 * @returns {string}
+	 */
+	makeIconPath() {
+		if (!this.public || this.isOpensource) {
```

Comment:
Here was the bug in previous implementation (code moved from AppsProvider), where Opensource apps icons was missing.

Root cause: It tried to use ".public.png" icon file for Opensource apps, but public file was not exist for opensource apps.

Solution: Added `|| this.isOpensource` into condition.